src/ast/ast.c

```diff
@@ -248,9 +248,15 @@ bool AST_ClauseContainsAggregation(const cypher_astnode_t *clause) {
 
 	void *value;
 	tm_len_t len;
-	char *funcName;
+	char *ptr;
+	char funcName[32];
 	TrieMapIterator *it = TrieMap_Iterate(referred_funcs, "", 0);
-	while(TrieMapIterator_Next(it, &funcName, &len, &value)) {
+	while(TrieMapIterator_Next(it, &ptr, &len, &value)) {
+		assert(len < 32);
+		// Copy the triemap key so that we can safely add a terinator character
+		memcpy(funcName, ptr, len);
```

Comment:
Looking at 
_consume_function_call_expression
`TrieMap_Add(referred_funcs, (char *)func_name, strlen(func_name), NULL, TrieMap_DONT_CARE_REPLACE);`
I think we can conclude that if `strlen(func_name)` is greater than 32 or some other value we agree upon then the function doesn't exists. if we'll do that we can avoid performing memcpy within `AST_ClauseContainsAggregation` and just work with a stack based buffer.

Reply:
We are using a stack based buffer, but we also need the `memcpy`. The reason is that writing a NULL terminator into the space returned by `TrieMapIterator_Next` is literally writing data into a part of the triemap that does not belong to this string. It's possible to overwrite critical bytes.

The alternative is to rewrite `Agg_FuncExists` and `Ar_FuncExists` to accept a non-terminated string and a length argument, but that gets a little thorny, since `Ar_FuncExists` invoked `_ToLower`, etc.